unit-tests/post-processing/test-post-processing-from-bag.py

```diff
@@ -0,0 +1,126 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#temporary fix to prevent the test from running on Win_SH_Py_DDS_CI 
+#test:donotrun:dds
+
+import pyrealsense2 as rs
+from rspy import test, repo
+import os.path
+import time
+################################################################################################
+pre_aligned_frames_map ={}
+frames_data_map = {}
+playback_status = None
+sensors = []
+align_from = rs.stream(rs.stream.depth)
+align = rs.align(rs.stream.color)
+
+
+def playback_callback(status):
+    global playback_status
+    playback_status = status
+
+
+def validate_ppf_results(result_frame_data, reference_frame_data):
+    result_bytearray_frame_data, result_profile = result_frame_data
+    reference_bytearray_frame_data, reference_profile = reference_frame_data
+
+    test.check_equal(result_profile.width(), reference_profile.width())
+    test.check_equal(result_profile.height(), reference_profile.height())
+    test.check_equal_lists(result_bytearray_frame_data,reference_bytearray_frame_data)
+
+
+def process_frame(frame, frame_source):
+    sensor_name = rs.sensor.from_frame(frame).get_info(rs.camera_info.name)
+    if sensor_name in pre_aligned_frames_map:
+        pre_aligned_frames_map[sensor_name].append(frame)
+    else:
+        pre_aligned_frames_map[sensor_name] = [frame]
+
+    if len(pre_aligned_frames_map[sensor_name]) == 2:
+        frameset = frame_source.allocate_composite_frame(pre_aligned_frames_map[sensor_name])
+        fs_align = align.process(frameset.as_frameset()).first_or_default(align_from)
+        fs_align_data = bytearray(fs_align.get_data())
+        fs_align_profile = fs_align.get_profile().as_video_stream_profile()
+        frames_data_map[sensor_name] = (fs_align_data, fs_align_profile)
+
+
+def load_ref_frames_to_map(frame):
+    if len(frames_data_map) < len(sensors):
+        sensor_name = rs.sensor.from_frame(frame).get_info(rs.camera_info.name)
+        frames_data_map[sensor_name] = (bytearray(frame.get_data()), frame.get_profile().as_video_stream_profile())
+
+
+def get_frames(callback):
```

Comment:
I like this function, but not with the frame-list logic: it makes it less generic. Why should it know a frames_data_map, etc.? That logic should be in the caller of `playback_file`?

Reply:
Done.

Reply:
Done